sn_networking/src/event.rs

```diff
@@ -274,19 +274,22 @@ impl SwarmDriver {
                                         .swarm
                                         .behaviour_mut()
                                         .kademlia
-                                        .add_address(&peer_id, multiaddr);
+                                        .add_address(&peer_id, multiaddr.clone());
                                 }
                             }
 
-                            // If the peer supports AutoNAT, add it as server
-                            if info.protocols.iter().any(|protocol| {
-                                protocol.to_string().starts_with("/libp2p/autonat/")
-                            }) {
-                                let a = &mut self.swarm.behaviour_mut().autonat;
-                                // It could be that we are on a local network and have AutoNAT disabled.
-                                if let Some(autonat) = a.as_mut() {
-                                    for multiaddr in addrs {
-                                        autonat.add_server(peer_id, Some(multiaddr));
+                            // only add new autonat peers
+                            if !self.dialed_peers.contains(&peer_id) {
```

Comment:
I'm not sure I like the effect of this condition. If we dial the first bootstrap node, we might not add it as an AutoNAT server with this. Also, `dialed_peers` by definition should contain a peer we are adding, because we want to make sure it's reachable.